permission/basic/backend.go

```diff
@@ -0,0 +1,368 @@
+
+			case evtStatusChanged := <-chStatusChanged:
+				if ac, err := types.AcctInfoMap.GetAccount(evtStatusChanged.Account); ac != nil {
+					types.AcctInfoMap.UpsertAccount(evtStatusChanged.OrgId, ac.RoleId, evtStatusChanged.Account, ac.IsOrgAdmin, types.AcctStatus(int(evtStatusChanged.Status.Uint64())))
+				} else {
+					log.Info("error fetching account information", "err", err)
+				}
+			case <-stopChan:
+				log.Info("quit account Contr watch")
+				return
+			}
+		}
+	}()
+	return nil
+}
+
+func (b *Backend) ManageRolePermissions() error {
+	chRoleCreated := make(chan *bb.RoleManagerRoleCreated, 1)
+	chRoleRevoked := make(chan *bb.RoleManagerRoleRevoked, 1)
+
+	opts := &bind.WatchOpts{}
+	var blockNumber uint64 = 1
+	opts.Start = &blockNumber
+	contract := b.Contr
+
+	if _, err := contract.PermRole.RoleManagerFilterer.WatchRoleCreated(opts, chRoleCreated); err != nil {
+		return fmt.Errorf("failed WatchRoleCreated: %v", err)
+	}
+
+	if _, err := contract.PermRole.RoleManagerFilterer.WatchRoleRevoked(opts, chRoleRevoked); err != nil {
+		return fmt.Errorf("failed WatchRoleRemoved: %v", err)
+	}
+
+	go func() {
+		stopChan, stopSubscription := ptype.SubscribeStopEvent()
+		defer stopSubscription.Unsubscribe()
+		for {
+			select {
+			case evtRoleCreated := <-chRoleCreated:
+				types.RoleInfoMap.UpsertRole(evtRoleCreated.OrgId, evtRoleCreated.RoleId, evtRoleCreated.IsVoter, evtRoleCreated.IsAdmin, types.AccessType(int(evtRoleCreated.BaseAccess.Uint64())), true)
+
+			case evtRoleRevoked := <-chRoleRevoked:
+				if r, _ := types.RoleInfoMap.GetRole(evtRoleRevoked.OrgId, evtRoleRevoked.RoleId); r != nil {
+					types.RoleInfoMap.UpsertRole(evtRoleRevoked.OrgId, evtRoleRevoked.RoleId, r.IsVoter, r.IsAdmin, r.Access, false)
+				} else {
+					log.Error("Revoke role - cache is missing role", "org", evtRoleRevoked.OrgId, "role", evtRoleRevoked.RoleId)
+				}
+			case <-stopChan:
+				log.Info("quit role Contr watch")
+				return
+			}
+		}
+	}()
+	return nil
+}
+
+func (b *Backend) ManageOrgPermissions() error {
+	chPendingApproval := make(chan *bb.OrgManagerOrgPendingApproval, 1)
+	chOrgApproved := make(chan *bb.OrgManagerOrgApproved, 1)
+	chOrgSuspended := make(chan *bb.OrgManagerOrgSuspended, 1)
+	chOrgReactivated := make(chan *bb.OrgManagerOrgSuspensionRevoked, 1)
+
+	opts := &bind.WatchOpts{}
+	var blockNumber uint64 = 1
+	opts.Start = &blockNumber
+	contract := b.Contr
+
+	if _, err := contract.PermOrg.OrgManagerFilterer.WatchOrgPendingApproval(opts, chPendingApproval); err != nil {
+		return fmt.Errorf("failed WatchNodePendingApproval: %v", err)
+	}
+
+	if _, err := contract.PermOrg.OrgManagerFilterer.WatchOrgApproved(opts, chOrgApproved); err != nil {
+		return fmt.Errorf("failed WatchNodePendingApproval: %v", err)
+	}
+
+	if _, err := contract.PermOrg.OrgManagerFilterer.WatchOrgSuspended(opts, chOrgSuspended); err != nil {
+		return fmt.Errorf("failed WatchNodePendingApproval: %v", err)
```

Comment:
Mistake in error message - should be "WatchOrgSuspended"

Reply:
done